### PR TITLE
Fix flickering country list issue

### PR DESCRIPTION
**Pull request by stefaniacardenas**
# Problem
Fix flickering country list on SMS country dropdown.

# Solution
The fix involves updating `deviceHasCameraSupport` in the Redux store only when the value changes. 

Bonus: Update component to use hooks.

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1411-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1411-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1411-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1411-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Capture/withCameraDetection.tsx [hunk 1]
```diff
@@ -10,49 +10,40 @@ type Props = ReduxProps & WithCameraDetectionProps
...
+        })
 
-    private cameraCheckerIntervalId?: number
-
-    componentDidMount() {
       // HACK: use of isSafari131 util function is a workaround specifically for
       //       Safari 13.1 bug that incorrectly returns a "videoinput" as "audioinput"
       //       on subsequent calls to enumerateDevices
       //       see https://bugs.webkit.org/show_bug.cgi?id=209580
       if (!isSafari131()) {
```
Comment (ethan605) [inline src/components/Capture/withCameraDetection.tsx:33]: Out of scope: this wasn't covered by unit tests. I'd be great if you can add some tests to cover it (please ignore it if you need this PR to be merged quickly)
swappy-20210416-110046

**Review by ethan605 — APPROVED** [review]
LGTM!

File: src/components/Capture/withCameraDetection.tsx (hunk 1 above)
Comment (kopijunkie) [reply src/components/Capture/withCameraDetection.tsx:33]: Agree with above point, but we should create a ticket to make sure it's done if it won't be covered in this PR.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1411-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Capture/withCameraDetection.tsx [hunk 2]
```diff
@@ -10,49 +10,40 @@ type Props = ReduxProps & WithCameraDetectionProps
...
-
-      // while checking if we have a camera or not, don't render anything
-      // otherwise we'll see a flicker, after we do work out what's what
-      if (hasCamera === null) return null
-
-      return (
-        <WrappedComponent {...this.props} hasCamera={this.state.hasCamera} />
-      )
-    }
+    }, [hasCamera])
```
[the remark this answers is not in the document]
Comment (kopijunkie) [reply src/components/Capture/withCameraDetection.tsx:41]: I've had to use this in a few places recently also. I tried out alternative that wouldn't require me using this `// eslint-disable-line react-hooks/exhaustive-deps` comment but found that didn't play nice. 😅

**Review by kopijunkie — APPROVED** [review]

File: src/components/Capture/withCameraDetection.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ethan605) [reply src/components/Capture/withCameraDetection.tsx:41]: IMO this check is pretty neat: by default, we should exhaustively add all the dependencies to the list. But in some special cases (like this), putting an `eslint-disable-line` here indicates that we _intended_ to do that. I often commentted to explain why we didn't need / couldn't add all the related dependencies.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1411-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1411-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Capture/withCameraDetection.tsx (hunk 1 above)
Comment (stefaniacardenas) [reply src/components/Capture/withCameraDetection.tsx:33]: I agree, will create a ticket as I believe we want to start the patch release process today

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1411-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**